names/names.go

```diff
@@ -210,149 +198,19 @@ const (
 	EUISOEWest1RegionID = "eu-isoe-west-1" // EU ISOE West.
 )
 
-var allRegionIDs = []string{
-	AFSouth1RegionID,
-	APEast1RegionID,
-	APNortheast1RegionID,
-	APNortheast2RegionID,
-	APNortheast3RegionID,
-	APSouth1RegionID,
-	APSouth2RegionID,
-	APSoutheast1RegionID,
-	APSoutheast2RegionID,
-	APSoutheast3RegionID,
-	APSoutheast4RegionID,
-	APSoutheast5RegionID,
-	CACentral1RegionID,
-	CAWest1RegionID,
-	EUCentral1RegionID,
-	EUCentral2RegionID,
-	EUNorth1RegionID,
-	EUSouth1RegionID,
-	EUSouth2RegionID,
-	EUWest1RegionID,
-	EUWest2RegionID,
-	EUWest3RegionID,
-	ILCentral1RegionID,
-	MECentral1RegionID,
-	MESouth1RegionID,
-	SAEast1RegionID,
-	USEast1RegionID,
-	USEast2RegionID,
-	USWest1RegionID,
-	USWest2RegionID,
-	CNNorth1RegionID,
-	CNNorthwest1RegionID,
-	USGovEast1RegionID,
-	USGovWest1RegionID,
-	USISOEast1RegionID,
-	USISOWest1RegionID,
-	USISOBEast1RegionID,
-	EUISOEWest1RegionID,
-}
-
-func Regions() []string {
-	return slices.Clone(allRegionIDs)
-}
-
-func DNSSuffixForPartition(partition string) string {
-	switch partition {
-	case "":
-		return ""
-	case ChinaPartitionID:
-		return "amazonaws.com.cn"
-	case ISOPartitionID:
-		return "c2s.ic.gov"
-	case ISOBPartitionID:
-		return "sc2s.sgov.gov"
-	case ISOEPartitionID:
-		return "cloud.adc-e.uk"
-	case ISOFPartitionID:
-		return "csp.hci.ic.gov"
-	default:
-		return "amazonaws.com"
+// PartitionForRegion returns the partition for the given Region.
+// Returns the empty partition if the Region is empty.
+// Returns the standard partition if no known partition includes the Region.
+func PartitionForRegion(region string) endpoints.Partition {
```

Comment:
Not a blocker, but consider moving this logic to `endpoints` or moving this out of `names`: it's not a "name", so the package is not semantically correct

Reply:
Will consider when I do the equivalent for the Region constants.